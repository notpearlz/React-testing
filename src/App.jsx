import { useState } from "react";

const App = ()=>{
  const [heading, setheading] = useState("Magnificent Rhinos")

  const clickHandler = (e)=>{
    e.preventDefault();

    setheading("Radical Monkeys")
  }
  return (
    <>
      <h1>{heading}</h1>
      <button type='button' onClick={clickHandler}>Click Me </button>
    </>
  );
}


export default App;