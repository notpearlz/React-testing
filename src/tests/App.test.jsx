import { describe, it, expect } from "vitest";
import { render, screen } from '@testing-library/react';
import App from '../App.jsx'
import userEvent from "@testing-library/user-event";

describe('App Component', () => {
  it('Renders magnificent rhinos', ()=>{
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });


  it('renders radical Manificent Rhinos after button click', async () => {
    const user = userEvent.setup();
    
    render(<App />)
    const button = screen.getByRole("button", {name: "Click Me"});

    await user.click(button);

    expect(screen.getByRole("heading").textContent).toMatch("Radical Monkeys");

  });



})


