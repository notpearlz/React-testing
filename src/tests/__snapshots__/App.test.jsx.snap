// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App Component > Renders magnificent rhinos 1`] = `
<div>
  <h1>
    Magnificent Rhinos
  </h1>
  <button
    type="button"
  >
    Click Me 
  </button>
</div>
`;
